### DISCUSSION TITLE
Redis session dies after idle for redis_strategy

### DISCUSSION THREAD

**Discussion by bert2002** [fastapi-users/fastapi-users#1313] [General]
Hi,
I am using the `get_redis_strategy` to store the sessions in redis and what I noticed is that after the process is running for some time (hours) and no one logged in/out the redis session times out. When this happens and someone tries to login, the user will see a server error (because the session died) and only after the process restarted, it will work again.

I am using the asyncio like this:
```
redis = redis.asyncio.from_url(os.getenv("REDIS_URL", None), decode_responses=True, max_connections=10)

async def get_redis_strategy() -> RedisStrategy:
  return RedisStrategy(await redis, lifetime_seconds=86400)

auth_backend = AuthenticationBackend(
  name="cookie",
  transport=cookie_transport,
  get_strategy=get_redis_strategy,
)
```
Does anyone has a suggestion on how to use a pool that will use a new connection for each `get_redis_strategy` after a session expired or should I approach this differently?

Thanks,
bert

**Comment by frankie567** [comment]
I'm surprised you see this kind of errors, since the Redis Py client gracefully handle disconnected connections and automatically reconnects: https://github.com/redis/redis-py/issues/932

Are you sure the error comes from a disconnected connection? Or maybe something in-between (like a network gateway) prevents the client from reconnecting? The exception and stacktrace you're getting could be useful to understand what happens exactly.

Anyway, there is something surprising in the code snippet you shared:

```py
return RedisStrategy(await redis, lifetime_seconds=86400)
```

the `await` is not necessary here (maybe harmful?) since `redis` is a plain `Redis` object. It should be:

```py
return RedisStrategy(redis, lifetime_seconds=3600)
```

**Reply by bert2002** [reply]
Hi @frankie567, just got to it to finally test this and you were right. The `await` not good. Once I removed it I didn't experience the problem anymore.
